api/nodecontrol/client_test.go

```diff
@@ -0,0 +1,943 @@
+package nodecontrol
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"disorder.dev/shandler"
+	"github.com/nats-io/nats-server/v2/server"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+	"github.com/synadia-io/nex/api/nodecontrol/gen"
+	"github.com/synadia-io/nex/models"
+	"github.com/synadia-io/nex/node"
+)
+
+const (
+	// NCUU2YIYXEPGTCDXDKQR7LL5PXDHIDG7SDFLWKE3WY63ZGCZL2HKIAJT - pub key
+	Node1ServerSeed string = "SNAB2T3VG2363NDA2JK7NT5O3FN5VCXI2MYJHOPFO2NIDXQU6DIWQTBQC4"
+	// XAL54S5FE6SRPONXRNVE4ZDAOHOT44GFIY2ZW33DHLR2U3H2HJSXXRKY - pub xkey
+	Node1XKeySeed string = "SXAOUP7RZFW5QPE2GDWTPABUDM5UIAK6BPULJPWZQAFFL2RZ5K3UYWHYY4"
+)
+
+func buildTestBinary(t testing.TB, binMain string, workingDir string) (string, error) {
```

Comment:
For test helper functions, fatal errors should not be returned and instead handled in the function. For example, if the `go build` command fails below, you can call `t.Fatal` directly (or if you introduce a `test.NoError` helper function). This avoids for the calling function (usually the actual test) to have to handle it.

Reply:
[126eead](https://github.com/synadia-io/nex/pull/473/commits/126eeadbf2414db6e2baa1c6eabd7f6c46bb250c) brings in [github.com/carlmjohnson/be](http://github.com/carlmjohnson/be)